### PR TITLE
Fixes #25010: Subscriptions page UI elements reflect user permissions

### PR DESCRIPTION
**Pull request by jeremylenz**
* Modify API endpoint `katello/api/v2/subscriptions` to add param `include_permissions`
* Check current user permissions and show/hide UI elements based on whether the user has permission to do that action.
Subscriptions list:
* `manage_subscription_allocations` - show edit controls for entitlements; Add Subscriptions button
Manage Manifest modal: 
* `import_manifest` - refresh or change manifest
* `delete_manifest` - delete a manifest
* `edit_organizations` - change/view Red Hat CDN URL

### CODE REVIEW

File: app/views/katello/api/v2/subscriptions/permissions.rabl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+collection @subscriptions
+
+if params.key?(:include_permissions)
+  user = User.current
```
Comment (akofink) [inline app/views/katello/api/v2/subscriptions/permissions.rabl]: Why not use [`current_user`](https://github.com/Katello/katello/blob/master/app/controllers/katello/concerns/api/api_controller.rb#L14)?

**Comment by waldenraines** [comment]
@jeremylenz is this a WIP or are you planning on squashing the commits?

**Comment by jeremylenz** [comment]
@waldenraines I was talking with @akofink about how to use the Github squash merge feature instead of command-line rebase, and was just testing to see what's possible..

**Comment by jeremylenz** [comment]
@akofink btw, switched back to `User.current` because `current_user` didn't work.
```ruby
{"displayMessage"=>"undefined local variable or method `current_user' for #<Rabl::Engine:0x000000001474d3f8>", "errors"=>["undefined local variable or method `current_user' for #<Rabl::Engine:0x000000001474d3f8>"]} 
```

**Comment by jeremylenz** [comment]
[test katello]

**Pull request merged**